hub/core/experimental/cleanlab.py

```diff
@@ -0,0 +1,253 @@
+# TODO: Add error checking for the len(tensors) after appending label issues tensors
+# TODO: Error checking, e.g. if number of samples not equal to number of labels in the initial dataset,
+# TODO: Add support for tensorflow models
+# TODO: Add types for the arguments
+# TODO: Create custom config for pip install (e.g. !pip install hub[’cleanlab’])
+# TODO: Add cleanlab kwargs
+# TODO: Add support to create valid and loss accuracy graphs
+# TODO: Add suppport for bounding boxes, task = 'classification' or 'segmentation'
+# TODO: Add tests for the functions
+# TODO: Use y as when calling fit() instead of passing the whole ds?
+# TODO: Use scikit learn shuffle for train set after getting the indices of train and holdout sets.
+# TODO: Instead of deleting label group, just overwrite it at the end if overwrite = True.
+# TODO: Add append parameter. If True, append label issues tensors to the dataset. If False, return label issues as np array.
+# TODO: Add guessed label to tensors
+# TODO: Add progress bar for the training
+# TODO: Add flag branch to move to a different branch.
+
+def get_dataset_tensors(dataset, tensors, dataloader_train_params, overwrite, verbose):
+    """
+    This function returns the tensors of a dataset. If a list of tensors is not provided,
+    it will try to find them in the dataloader_train_params in the transform. If none of
+    these are provided, it will iterate over the dataset tensors and return any tensors
+    that match htype 'image' for images and htype 'class_label' for labels. Additionally,
+    this function will also check if the dataset already has a label_issues group.
+    """
+
+    tensors_list = list(dataset.tensors)
+
+    if (
+        "label_issues/is_label_issue" in tensors_list
+        or "label_issues/label_quality_scores" in tensors_list
+    ):
+        if overwrite:
+            if verbose:
+                print("Found existing label_issues tensor. Deleting the tensor...")
+            dataset.delete_group("label_issues")
+            dataset.commit("Removed label issues", allow_empty=True)
+
+        else:
+            raise ValueError(
+                "The group of tensors label_issues already exist. Use overwrite = True, to overwrite the label_issues tensors."
+            )
+
+    if tensors is not None:
+        tensors = map_tensor_keys(dataset, tensors)
+
+    # Try to get the tensors from the dataloader parameters.
+    elif (
+        dataloader_train_params
+        and "transform" in dataloader_train_params
+        and isinstance(dataloader_train_params["transform"], dict)
+    ):
+        tensors = map_tensor_keys(
+            dataset,
+            [k for k in dataloader_train_params["transform"].keys() if k != "index"],
+        )
+
+    # Map the images and labels tensors to the corresponding tensors in the dataset.
+    images_tensor, labels_tensor = None, None
+
+    if tensors:
+        for tensor in tensors:
+            if dataset[tensor].htype == "image":
```

Comment:
I'm curious if there can be any other htypes that I haven't covered here. What would be some edge cases I should look into more?